### DISCUSSION TITLE
How to adjust vercel.json to work with multiple Flask applications

### DISCUSSION THREAD

**Discussion by kasperBengtson** [vercel/community#4760] [Help]
### Summary

I am using Vercel's serverless functions for a couple of Flask apps. I've previously been inserting this in my vercel.json, according to the documentation:

    "rewrites": [
      { "source": "/(.*)", "destination": "/api/app" }
    ]
  
 However, I now have an app with multiple functions placed in subfolders of api/, for example /api/function_one/app.
 
 How do I make a http-request to this specific function?
 
 I tried extending the rewrite function like this:
 
     "rewrites": [
      { "source": "/function_one/(.*)", "destination": "/api/function_one/app" }
    ]
    
 But when I then make the request to https://vercelapp.com/function_one, it returns 404.
 
 The Flask route of the app is "/".
 
 For most of you, the answer is probably simple, so I hope you would like to help. I'm new in developing applications and especially deploying them. 

### Steps to Reproduce

Not available

**Comment by amyegan** [comment]
Hey @kasperBengtson! You should configure rewrites based on the deployed routes rather than the repo file path.

**Reply by kasperBengtson** [reply]
Thanks, @amyegan. Could you please show me an example using Pyton? That would be much appreciated!

**Reply by amyegan** [reply]
Since [rewrites](https://vercel.com/docs/projects/project-configuration#rewrites) use URL paths, the solution is the same regardless of the runtime. If I had a project that deployed a page or function at `example.com/api/test` but I wanted people to be able to see that page at `example.com/my-test` instead, I would use the rewrite config to let them visit the source `/my-test` and see the page as if they had visited the destination `/api/test`.

**Reply by kasperBengtson** [reply]
Thanks, @amyegan. What I want is to allow people to make a request to example.com/function_one, as if they where making a request to my Flask app which is located in api/function_one/app.py. It has the route: @app.route('/'). 

If I understand you correctly, I should change the flask route to @app.route('/function_one_route') and change the rewrite to:

 "rewrites": [
  { "source": "/function_one/(.*)", "destination": "/api/function_one/app/function_one_route" }

]
